plenum/test/node_request/test_unit_setup_for_non_master.py

```diff
@@ -0,0 +1,168 @@
+    replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] = deque()
+    for node in txnPoolNodeSet:
+        replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] \
+            .append((prepare, node.name))
+    replica._setup_last_ordered_for_non_master()
+    assert replica.last_ordered_3pc == (0, 0)
+
+
+def test_setup_last_ordered_for_non_master_without_prepare(txnPoolNodeSet,
+                                                           sdk_wallet_client):
+    inst_id = 1
+    replica = getNonPrimaryReplicas(txnPoolNodeSet, inst_id)[-1]
+    replica.preparesWaitingForPrePrepare.clear()
+    replica.prePreparesPendingPrevPP.clear()
+    replica.last_ordered_3pc = (0, 0)
+    timestamp = time.time()
+    ppSeqNo = 5
+    preprepare, prepare = \
+        _create_prepare_and_preprepare(inst_id,
+                                       replica.viewNo,
+                                       ppSeqNo,
+                                       timestamp,
+                                       sdk_wallet_client)
+    replica.prePreparesPendingPrevPP[replica.viewNo, ppSeqNo] = deque()
+    replica.prePreparesPendingPrevPP[replica.viewNo, ppSeqNo] \
+        .append((preprepare, replica.primaryName))
+    replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] = deque()
+    replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] \
+        .append((prepare, txnPoolNodeSet[-1].name))
+    replica._setup_last_ordered_for_non_master()
+    assert replica.last_ordered_3pc == (0, 0)
+
+
+def test_setup_last_ordered_for_non_master_for_master(txnPoolNodeSet,
+                                                      sdk_wallet_client):
+    inst_id = 0
+    replica = getNonPrimaryReplicas(txnPoolNodeSet, inst_id)[-1]
+    replica.preparesWaitingForPrePrepare.clear()
+    replica.prePreparesPendingPrevPP.clear()
+    replica.last_ordered_3pc = (0, 0)
+    timestamp = time.time()
+    ppSeqNo = 5
+    preprepare, prepare = \
+        _create_prepare_and_preprepare(inst_id,
+                                       replica.viewNo,
+                                       ppSeqNo,
+                                       timestamp,
+                                       sdk_wallet_client)
+    replica.prePreparesPendingPrevPP[replica.viewNo, ppSeqNo] = deque()
+    replica.prePreparesPendingPrevPP[replica.viewNo, ppSeqNo] \
+        .append((preprepare, replica.primaryName))
+    replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] = deque()
+    for node in txnPoolNodeSet:
+        replica.preparesWaitingForPrePrepare[replica.viewNo, ppSeqNo] \
+            .append((prepare, node.name))
+    replica._setup_last_ordered_for_non_master()
+    assert replica.last_ordered_3pc == (0, 0)
+
+
+def test_setup_last_ordered_for_non_master_without_catchup(txnPoolNodeSet,
+                                                           sdk_wallet_client):
+    inst_id = 1
+    last_ordered_3pc = (5, 12)
```

Comment:
Why does last_ordered_3pc view_no equal to 5 when the nodes are in view 0?

Reply:
We simulate the state of the working pool after some time after the start. node.ViewNo has not used in this case, but we can change in to 5 or  last_ordered_3pc = (0, 12) . It wil not influence to test.